Lesson4/step2/project.py

```diff
@@ -29,12 +32,10 @@
 
 # Create anti-forgery state token
 @app.route('/login')
-def showLogin():
-    state = ''.join(random.choice(string.ascii_uppercase + string.digits)
-                    for x in xrange(32))
+def show_login():
     login_session['state'] = state
     # return "The current session state is %s" % login_session['state']
-    return render_template('login.html', STATE=state)
+    return render_template('login.html', STATE=login_session['state'], google_web=google_web, facebook_web=facebook_web)
```

Comment:
Return the imported variables so that they can be expanded without having to hard-code values.